### PR TITLE
[GEOT-6463] Remove support for json path in MongoDB module

### PR DESCRIPTION
**Pull request by taba90**
This pull request has a related Jira Ticket https://osgeo-org.atlassian.net/browse/GEOT-6463

### CODE REVIEW

File: modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/complex/JsonSelectFunction.java [hunk 1]
```diff
@@ -52,11 +52,9 @@ public Object evaluate(Object object) {
         String path = (String) this.params.get(0).evaluate(object);
         if (object instanceof MongoCollectionFeature) {
             // in the case of a mongo collection we need to append the parent path
-            if (!MongoComplexUtilities.useLegacyPaths()) {
                 String parentPath = ((MongoCollectionFeature) object).getCollectionPath();
```
Comment (nmco) [inline modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/complex/JsonSelectFunction.java]: Looks the spacing was not fixed, let's see if the build complains.

Comment (taba90) [reply modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/complex/JsonSelectFunction.java]: fixed thanks

**Comment by nmco** [comment]
For reference, mailing list discussion:
- http://osgeo-org.1560.x6.nabble.com/Remove-legacy-json-path-support-td5424988.html

**Comment by taba90** [comment]
Travis failing seems unrelated to this pr

**Comment by nmco** [comment]
@taba90 can you copy paste the actual error?

**Comment by nmco** [comment]
Indeed looks unrleated:
```
[ERROR] Tests run: 18, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.083 s <<< FAILURE! - in org.geotools.filter.text.ecql.ECQLTemporalPredicateTest
[ERROR] beforeFilterWithOgcInstantToEcql(org.geotools.filter.text.ecql.ECQLTemporalPredicateTest)  Time elapsed: 0.023 s  <<< FAILURE!
org.junit.ComparisonFailure: expected:<... 2019-12-12T11:29:40[.000]+00:00> but was:<... 2019-12-12T11:29:40[]+00:00>
	at org.geotools.filter.text.ecql.ECQLTemporalPredicateTest.beforeFilterWithOgcInstantToEcql(ECQLTemporalPredicateTest.java:168)
```

I don't think a change on MongoDB App-Schema integration has an impact on:
```
[INFO] OGC CQL to Filter parser ........................... FAILURE [  5.257 s]
```

**Pull request merged**